hydromt_imod/imod.py

```diff
@@ -64,71 +66,316 @@ def __init__(
             logger=logger,
         )
 
+    def _raster_or_constant(
+        self,
+        value: Union[int, float, None],
+        fn: Union[str, Path, None],
+        like: xr.DataArray,
+        method: str = "bilinear",
+    ):
+        if not ((value is None) ^ (fn is None)):
+            raise ValueError(
+                "Either provide a constant or a filename."
+                f"constant: {value}, filename: {fn}"
+            )
+
+        elif value is not None:
+            return self._prepare_constant(value, like)
+        elif fn is not None:
+            return self._prepare_raster(fn, like, method=method)
+
+    def _prepare_constant(self, value, like):
+        return xr.full_like(like, fill_value=value)
```

Comment:
I had never realized xarray had a full_like method.. 
HydroMT also has a [raster.full_like](https://deltares.github.io/hydromt/latest/_generated/hydromt.raster.full_like.html) method which takes care of the passing the crs and nodata attributes but does not have the possibility to add a fill_value (we just use the nodata value to fill the raster)

@DirkEilander shall we maybe update the raster.full_like function to also have a fill_value argument? This way we would be more in-line with xarray and here we could call 

```suggestion
        return hydromt.raster.full_like(like, fill_value=value, nodata=nodata)
```